### PR TITLE
[DF] Improve RNode.C tutorial

### PR DESCRIPTION
**Pull request by eguiraud**
Show a few more cool tricks, in particular building computation graphs
using C++ control flow.

This also removes a memleak of the return value of
TClassEdit::DemangleName (thanks for pointing it out Sergei!).

### CODE REVIEW

File: tutorials/dataframe/df025_RNode.C [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
+/// Luckily, every RDF object can be converted to the generic RNode type.
+/// This tutorial shows how to take advantage of RNode to easily manipulate RDataFrames.
 ///
 /// \macro_code
 /// \macro_output
 ///
-/// \date December 2018
+/// \date June 2020
 /// \author Danilo Piparo
+/// \author Enrico Guiraud
```
Comment (hageboeck) [inline tutorials/dataframe/df025_RNode.C:14]: If I remember correctly, doxygen wants `\authors ..., ...`. Not sure, though. Olivier could know this. You would have to check how this typesets once doxygen has been regenerated.

File: tutorials/dataframe/df025_RNode.C [hunk 2]
```diff
@@ -1,38 +1,43 @@
...
 
-/// This function does not need to be a template: the RNode type accommodates all
-/// possible nodes.
+/// A generic function that takes an RDF and applies a string filter
```
Comment (hageboeck) [inline tutorials/dataframe/df025_RNode.C:16]:
```suggestion
/// A generic function that takes an RDF node and applies a string filter
```

File: tutorials/dataframe/df025_RNode.C [hunk 3]
```diff
@@ -1,38 +1,43 @@
 /// \file
 /// \ingroup tutorial_dataframe
 /// \notebook
-/// RNode is a generic type which represents any transformation node in the computation graph.
-/// This tutorial shows how to take advantage of the RNode class.
+/// Each RDataFrame object has its own type. It helps with performance,
```
Comment (hageboeck) [inline tutorials/dataframe/df025_RNode.C:4]: Add a brief docstring that doxygen can show [here](https://root.cern/doc/master/group__tutorial__dataframe.html).
```suggestion
/// Manipulating computation graphs with RNode.
///
/// Each RDataFrame object has its own type. It helps with performance,
```

File: tutorials/dataframe/df025_RNode.C [hunk 4]
```diff
@@ -1,38 +1,43 @@
...
+   auto maybe_filtered = ROOT::RDF::RNode(f2);
+   if (someCondition)
+      maybe_filtered = maybe_filtered.Filter("rdfentry_ > 3");
+
+   // Adding new columns with Define in a loop is simple thanks to ROOT::RDF::RNode
+   auto with_columns = ROOT::RDF::RNode(maybe_filtered);
+   for (auto i = 0; i < 3; ++i)
+      with_columns = with_columns.Define("x" + std::to_string(i), "42");
+
+   // RNode's can be used exactly like any other RDF object
```
Comment (hageboeck) [inline tutorials/dataframe/df025_RNode.C:41]:
```suggestion
   // RNodes can be used exactly like any other RDF object
```

File: tutorials/dataframe/df025_RNode.C (hunk 1 above)
Comment (eguiraud) [reply tutorials/dataframe/df025_RNode.C:14]: if it's wrong, there are a lot of places that we need to fix :D

**Pull request merged**